specification/ingest/_types/Processors.ts

```diff
@@ -1046,6 +1053,42 @@ export class PipelineProcessor extends ProcessorBase {
   ignore_missing_pipeline?: boolean
 }
 
+export class RedactProcessor extends ProcessorBase {
+  /**
+   * The field to be redacted
+   */
+  field: Field
+  /**
+   * A list of grok expressions to match and redact named captures with
+   */
+  patterns: GrokPattern[]
+  /*
+   * A map of pattern-name and pattern tuples defining custom patterns to be used by the processor.
+   * Patterns matching existing names will override the pre-existing definition
+   */
+  pattern_definitions?: Dictionary<string, string>
```

Comment:
Should the value part be of type `GrokPattern` as well here?

Reply:
No, because they're not grok patterns, see this example: https://www.elastic.co/guide/en/elasticsearch/reference/current/grok-processor.html#custom-patterns